src/components/ContentUploader/OverallUploadsProgressBar.js

```diff
@@ -16,51 +16,54 @@ type Props = {
     percent: number,
     onClick: Function,
     onKeyDown: Function,
-    view: View
+    view: View,
+    numberOfItems: number
 };
 
-/**
- * Get upload status
- *
- * @param {View} view
- * @return {FormattedMessage|string}
- */
-const getUploadStatus = (view) => {
-    switch (view) {
-        case VIEW_UPLOAD_IN_PROGRESS:
-            return <FormattedMessage {...messages.uploadsManagerUploadInProgress} />;
-        case VIEW_UPLOAD_SUCCESS:
-        case VIEW_UPLOAD_EMPTY:
-            return <FormattedMessage {...messages.uploadsManagerUploadComplete} />;
-        case VIEW_ERROR:
-            return <FormattedMessage {...messages.uploadsManagerUploadFailed} />;
-        default:
-            return '';
-    }
-};
+const OverallUploadsProgressBar = ({ percent, view, onClick, onKeyDown, isVisible, numberOfItems }: Props) => {
+    /**
+     * Get upload status
+     *
+     * @param {View} view
+     * @return {FormattedMessage|string}
+     */
+    const getUploadStatus = () => {
```

Comment:
Any reason this wasn't fine like it was before outside? Generally nesting is fine for creating closures, but if a function can be pure by being outside with a couple of args, I think it might be better? Helps with reduction of horizontal indentation too, aka arrowness. And then can be exported too for testing, that is its more testable.

So in the case of numberOfItems === 0, you would pass in isEmpty to getPercent(view, percent, isEmpty)

Reply:
I don't have strong opinions on this. The only motivation was to avoid passing around args. I don't think I'll test these UI methods individually, probably will be taking a bunch of snapshots by variating props. But if you say the box-ui-elements style guide prefers pure functions, I'll happily change the implementation back to the way it was.

isEmpty 👍